File: Day_4/Sliding_Window/code/Solution_5/sol.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        if (k <= 1) return 0;

        int left = 0, result = 0;
        long long product = 1;

        for (int right = 0; right < nums.size(); ++right) {
            product *= nums[right];

            while (product >= k) {
                product /= nums[left++];
            }

            result += (right - left + 1);
        }

        return result;
    }
};

File: Day_4/Sliding_Window/code/Solution_4/sol.java
package Day_4.Sliding_Window.code.Solution_4;

public class sol {
    public int longestOnes(int[] nums, int k) {
        int left = 0, right = 0, maxLen = 0, zeros = 0;

        while (right < nums.length) {
            if (nums[right] == 0) zeros++;

            while (zeros > k) {
                if (nums[left] == 0) zeros--;
                left++;
            }

            maxLen = Math.max(maxLen, right - left + 1);
            right++;
        }

        return maxLen;
    }
}
